src/lib/bootloader/mbr_update.rb

```diff
@@ -88,12 +88,32 @@ def install_generic_mbr
     end
 
     def set_parted_flag(disk, part_num, flag)
+      # we need at first clear this flag to avoid multiple flags (bnc#848609)
+      reset_flag(disk, flag)
+
+      # and then set it
       command = "/usr/sbin/parted -s #{disk} set #{part_num} #{flag} on"
       out = Yast::WFM.Execute(Yast::Path.new(".local.bash_output"), command)
       log.info "Command `#{command}` output: #{out}"
       out
     end
 
+    def reset_flag(disk, flag)
+      command = "/usr/sbin/parted -s #{disk} print"
+      out = Yast::WFM.Execute(Yast::Path.new(".local.bash_output"), command)
+      log.info "Command `#{command}` output: #{out}"
+      return if out["exit"] != 0
+
+      partitions = out["stdout"].lines.grep(/\s#{flag}/)
```

Comment:
This is dangerous since a user can set an arbitrary partition name on GPT, so the output could be:
<pre>Number  Start   End    Size   File system  Name         Flags
 1      1049kB  822MB  821MB  btrfs        legacy_boot  boot</pre>